app/controllers/clients_controller.rb

```diff
@@ -21,4 +26,13 @@ def client_params
     params.fetch(:client, {})
       .permit(:first_name, :last_name, :birth_date, :phone_number, :active)
   end
+
+  def sorted_clients
+    # sort clients with unread messages to the top,
+    # no matter when they were last contacted
+    current_user.clients.all.sort{
+      |c1, c2| [c2.unread_message_count, c2.contacted_at] <=> [c1.unread_message_count, c1.contacted_at]
+    }
```

Comment:
Fun ruby tip, you can use `Enumerable#sort_by` to be a bit more efficient because the sort block will be called only once per item instead of this block being called n^2 times.

```ruby
current_user.cilents.all.sort_by { |c| [c.unread_message_count, c.contacted_at] }
```